src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/model/ccd/mappers/SupplementaryEvidenceMapper.java

```diff
@@ -13,7 +18,20 @@ public SupplementaryEvidenceMapper() {
         // empty mapper construct
     }
 
-    public SupplementaryEvidence mapEnvelope(Envelope envelope) {
-        return new SupplementaryEvidence(mapDocuments(envelope.documents));
+    public SupplementaryEvidence map(List<Document> existingDocs, List<Document> envelopeDocs) {
+
+        Stream<Document> docsToAdd =
+            envelopeDocs
+                .stream()
+                .filter(d -> existingDocs.stream().noneMatch(e -> Objects.equals(e.url, d.url)));
```

Comment:
should the check for dcn be here too?

Reply:
just the url, it uniquely identifies the file

Reply:
ah, the description in the ticket now have changed